### ISSUE TITLE
flutter run --profile on iOS doesn't have profiling turned on in observatory?

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#4974]
I just ran `flutter -d i run --profile` and then I opened Observatory to look at a CPU profile. I see this:

![screen shot 2016-07-19 at 6 18 06 pm](https://cloud.githubusercontent.com/assets/5479/16972009/2e68c72a-4ddd-11e6-9b5a-f1ae8563abbb.png)

"""
Profiling is disabled

Perhaps the profile flag has been disabled for this VM. 
"""

and

```
Modified Flags

// Run optimizing compilation in background
background_compilation  =  false

// Ignore unrecognized flags.
ignore_unrecognized_flags  =  true

// Time between profiler samples in microseconds. Minimum 50.
profile_period  =  1000

// Enable the profiler.
profiler  =  false
```

**Comment by rmacnak-google** [comment]
flutter/engine@3a4e0a0c80c71b7cf122b3c36d59828ab222184c

**Comment by sethladd** [comment]
Thanks @rmacnak-google !

**Comment by sethladd** [comment]
Curious, how long until this takes effect?

I see https://github.com/flutter/engine/pull/2804 landed yesterday, but I just tried it, and still no profiling.

**Comment by rmacnak-google** [comment]
The change was reverted because some flows were broken. It seems XCode attaches lldb by default to simulators. The likely fix is have profiling disabled by default and have `flutter run --profile` explicitly enable it.

**Comment by sethladd** [comment]
Confirmed, `--profile` isn't enabling CPU profiling on iOS hardware.

**Comment by Hixie** [comment]
This seems like a serious limitation of our iOS workflow. We definitely want to be able to profile both --debug (default) and --profile apps, like we can on Android.

**Comment by eseidelGoogle** [comment]
We'd have to verify if this was still true.  @xster or @cbracken might cross this in their iOS explorations.

**Comment by rmacnak-google** [comment]
Fixed by flutter/engine@327d122de9eaf3739996610a3630408bdcc52103 and flutter/flutter@4efb4282549e4b1d5379a87370d5891a1a4d746e.
